core/client/app/components/gh-profile-image.js

```diff
@@ -0,0 +1,86 @@
+/* global md5 */
+import Ember from 'ember';
+import {request as ajax} from 'ic-ajax';
+
+export default Ember.Component.extend({
+    email: '',
+    size: 90,
+    image: null,
+
+    ghostPaths: Ember.inject.service('ghost-paths'),
+
+    hasUploadedImage: Ember.computed.notEmpty('data'),
+
+    defaultImage: Ember.computed('ghostPaths', function () {
+        var url = this.get('ghostPaths.url').asset('/shared/img/user-image.png');
+        return `background-image: url(${url})`.htmlSafe();
+    }),
+
+    imageBackground: Ember.computed('email', function () {
+        var email = this.get('email') || '',
+            size = this.get('size'),
+            url = 'http://www.gravatar.com/avatar/' + md5(email) + '?s=' + size + '&d=blank';
+        return `background-image: url(${url})`.htmlSafe();
+    }),
+
+    didInsertElement: function () {
+        var self = this,
+            size = this.get('size'),
+            uploadElement = Ember.$('.js-file-input');
+
+        uploadElement.fileupload({
+            url: self.get('ghostPaths.url').api('uploads'),
+            dropZone: Ember.$('.js-img-dropzone'),
+            previewMaxHeight: size,
+            previewMaxWidth: size,
+            previewCrop: true,
+            maxNumberOfFiles: 1,
+            autoUpload: false,
+            acceptFileTypes: /(\.|\/)(gif|jpe?g|png|svg?z)$/i
+        })
+        .on('fileuploadadd', Ember.run.bind(self, self.queueFile))
+        .on('fileuploadprocessalways', Ember.run.bind(self, self.triggerPreview));
+    },
+
+    queueFile: function (e, data) {
+        this.set('data', data);
+    },
+
+    triggerPreview: function (e, data) {
+        var file = data.files[data.index];
+        if (file.preview) {
+            Ember.$('.js-img-preview').empty().append($(file.preview).addClass('gravatar-img'));
+        }
+    },
+
+    sendFile: function (user) {
+        var self = this,
+            image = this.get('data');
+
+        return new Ember.RSVP.Promise(function (resolve, reject) {
+            image.formData = {};
+            image.submit()
+                .success(function (response) {
+                    user.image = response;
+                    ajax({
+                        url: self.get('ghostPaths.url').api('users', user.id.toString()),
+                        type: 'PUT',
+                        data: {
+                            users: [user]
+                        }
+                    }).then(function () {
+                        resolve();
+                    }).catch(function (resp) {
+                        reject(resp);
+                    });
+                })
+                .error(function (jqXHR) {
+                    reject(jqXHR);
+                });
+        });
+    },
+
+    register: Ember.on('init', function () {
+        this.set('handle', this);
```

Comment:
The controller needs to call the sendFile function on the component, so a two-way binding I think is somewhat necessary, unless there's a better way